core/controllers/collection_viewer.py

```diff
@@ -33,22 +32,18 @@ class CollectionPage(base.BaseHandler):
     @acl_decorators.can_play_collection
     def get(self, collection_id):
         """Handles GET requests."""
+        collection_rights = rights_manager.get_collection_rights(
```

Comment:
Here and below: why not do this _after_ trying to retrieve the collection? Saves a datastore GET operation in the case where the collection does not exist.

Better still, see if you can find a clean way to batch this with the get_collection_by_id() (see how Arunabh does it in his learner dashboard PRs) so that only one datastore GET operation is required.